### ISSUE TITLE
Rename component.loaded & .mounted

### ISSUE THREAD

**Issue by st-pasha** [flame-engine/flame#1467]
> MIssing docs, but other than that it looks good!
I would probably have expected a `bool` from a field named `loaded`, but since we have the same with `mounted` we should follow that.

_Originally posted by @spydon in https://github.com/flame-engine/flame/pull/1466#pullrequestreview-914639988_

Given that the names `.loaded` and `.mounted` sound like they would return boolean flags, I suggest renaming those methods into `.whenLoaded` and `.whenMounted`. So, they could be called as
```dart
await component.whenLoaded;
await component.whenMounted;
```

**Comment by erickzanardo** [comment]
I disagree with this, `component.whenLoaded.when(...)` would sound very weird. `mounted` and `loaded` makes a perfect reading when put before an `await` keyword.

**Comment by st-pasha** [comment]
```dart
component.whenLoaded.then(() {
  ...
});
```
sounds quite ok though, at least better than `component.loaded.then`.

**Comment by erickzanardo** [comment]
This seems to be just based on personal opinion, `component.loaded.then` sounds good to me, not as good as `component.loaded.whenComplete` but it is fine for me.

I vote for not renaming those getters.

**Comment by st-pasha** [comment]
The original premise of this issue was the concern, raised by @spydon, that the name `.loaded` sounds ambiguous: as if it means a boolean flag, not a future. 

Do you believe that that concern is invalid, or do you think it's valid but the names that I suggested are not adequate?

**Comment by spydon** [comment]
It is the same with `Images.ready`, I would also have expected a boolean from the adjective there.
Are there any futures like this in Flutter which we can take the naming convention from?

**Comment by st-pasha** [comment]
> It is the same with Images.ready in #1441, I would also have expected a boolean from the adjective there.

Good point, there is a possibility of `ready` being confusing as well.

I guess one reason that it didn't look ambiguous to me, is that `.ready()` is a method, and I normally wouldn't expect a boolean indicator flag to be a method. So to me, `images.ready()` looks more like a future than `images.ready`.

Perhaps the same reasoning can be applied here as well? `component.loaded()` looks much less like a boolean than `component.loaded`.

**Comment by spydon** [comment]
I'll just repeat what I said on the other thread here too. Since [effective dart](https://dart.dev/guides/language/effective-dart/design#prefer-a-non-imperative-verb-phrase-for-a-boolean-property-or-variable) expects you to have `is`, `has`, `are` etc on fields that are boolean, then we could assume that these fields are not booleans.

> Perhaps the same reasoning can be applied here as well? component.loaded() looks much less like a boolean than component.loaded.

I think that might be clearer, I think the most important is that we follow the same structure on all of these kind of methods/fields.

**Comment by spydon** [comment]
Since they aren't actually doing much we decided it makes more sense to have them as getters, if you can make `ready()` a getter too we'll merge that too so that we get consistency over those methods. :)

**Comment by st-pasha** [comment]
@spydon I've double-checked with the Dart style guide, and here's what it has to say on the topic of [choosing between a getter or a method](https://dart.dev/guides/language/effective-dart/design#do-use-getters-for-operations-that-conceptually-access-properties):

> Even so, choosing a getter over a method sends an important signal to the caller. The signal, roughly, is that the operation is “field-like”. The operation, at least in principle, could be implemented using a field, as far as the caller knows. That implies:
> - The operation does not take any arguments and returns a result.
> - The caller cares mostly about the result. 
> - The operation does not have user-visible side effects.
> - The operation is idempotent. 
> - The resulting object doesn’t expose all of the original object’s state. 
>
> If all of the above describe your operation, it should be a getter. 

So, in this case `.ready()` is used primarily in order to create a side-effect, whereas its result is essentially `void`, and the user doesn't care about it all.

**Comment by spydon** [comment]
> So, in this case `.ready()` is used primarily in order to create a side-effect, whereas its result is essentially `void`, and the user doesn't care about it all.

The side effect is kind of already running in the background though? So I think that you could argue that this is only a getter for the state of the already running side effects caused by `load`.

**Comment by spydon** [comment]
I realized now that we also have the `FlameGame.ready()` method, that one feels more natural as a method though.

**Comment by erickzanardo** [comment]
But we are keeping mounted and loaded as they are right? So maybe we should close this?

**Comment by spydon** [comment]
> But we are keeping mounted and loaded as they are right? So maybe we should close this?

This issue is about `images.ready()` too, since it falls into the same category, and @st-pasha haven't changed that one to a getter yet.

**Comment by st-pasha** [comment]
We're being stuck in a [Fredkin's paradox](https://en.wikipedia.org/wiki/Fredkin%27s_paradox) here :)

I think one aspect here is what part of speech the name of the method/getter is. Say, if it's an imperative verb, then there should probably be parentheses, otherwise it just feels weird. If it's an adjective, then it could be either a getter or a method.

So, the question then is of the semantics of the name: do we intend `ready` to be a verb (as in "make this ready!"), or to be an adjective ("is this ready?"). For a verb, a method is more natural: `game.ready()`, meaning "make this game ready"; similarly `images.ready()` would mean "make all images ready". On the other hand, if we want to lean towards the adjective side of the word, then it could be a getter, but in that there should probably be some helper word to help disambiguate, like maybe `await images.whenReady`.

**Comment by spydon** [comment]
> We're being stuck in a [Fredkin's paradox](https://en.wikipedia.org/wiki/Fredkin%27s_paradox) here :)
> 
> I think one aspect here is what part of speech the name of the method/getter is. Say, if it's an imperative verb, then there should probably be parentheses, otherwise it just feels weird. If it's an adjective, then it could be either a getter or a method.
> 
> So, the question then is of the semantics of the name: do we intend `ready` to be a verb (as in "make this ready!"), or to be an adjective ("is this ready?"). For a verb, a method is more natural: `game.ready()`, meaning "make this game ready"; similarly `images.ready()` would mean "make all images ready". On the other hand, if we want to lean towards the adjective side of the word, then it could be a getter, but in that there should probably be some helper word to help disambiguate, like maybe `await images.whenReady`.

Let's keep the `ready`s as verbs and the others as adjectives then and we can merge your PR, I think that can make sense from what they do too.
